src/__tests__/Field.spec.js

```diff
@@ -1035,6 +1078,101 @@ const describeField = (name, structure, combineReducers, expect) => {
+      })
+      const passwordInput = createSpy(props => <input {...props.input}/>).andCallThrough()
+      const confirmInput = createSpy(props => <input {...props.input}/>).andCallThrough()
+      const warn = values => {
+        const password = getIn(values, 'password')
+        const confirm = getIn(values, 'confirm')
+        return password === confirm ? {} : { confirm: 'Should match. Or not. Whatever.' }
+      }
+      class Form extends Component {
+        render() {
+          return (<div>
+            <Field name="password" component={passwordInput}/>
+            <Field name="confirm" component={confirmInput}/>
+          </div>)
+        }
+      }
+      const TestForm = reduxForm({
+        form: 'testForm',
+        warn
+      })(Form)
+      TestUtils.renderIntoDocument(
+        <Provider store={store}>
+          <TestForm/>
+        </Provider>
+      )
+
+      // password input rendered
+      expect(passwordInput).toHaveBeenCalled()
+      expect(passwordInput.calls.length).toBe(1)
+
+      // confirm input rendered with warning
+      expect(confirmInput).toHaveBeenCalled()
+      expect(confirmInput.calls.length).toBe(1)
+      expect(confirmInput.calls[ 0 ].arguments[ 0 ].meta.warning).toBe('Should match. Or not. Whatever.')
+
+      // update password field so that they match
+      passwordInput.calls[ 0 ].arguments[ 0 ].input.onChange('redux-form rocks')
+
+      // password input rerendered
+      expect(passwordInput.calls.length).toBe(2)
+
+      // confirm input should also rerender, but with no warning
+      expect(confirmInput.calls.length).toBe(2)
+      expect(confirmInput.calls[ 1 ].arguments[ 0 ].meta.warning).toBe(undefined)
+    })
+
+    it('should rerender when sync warning is cleared', () => {
+      const store = makeStore()
+      const usernameInput = createSpy(props => <input {...props.input}/>).andCallThrough()
+      const warn = values => {
+        const username = getIn(values, 'username')
+        return username ? {} : { username: 'Recommended' }
+      }
+      class Form extends Component {
+        render() {
+          return (<div>
+            <Field name="username" component={usernameInput}/>
+          </div>)
+        }
+      }
+      const TestForm = reduxForm({
+        form: 'testForm',
+        warn
+      })(Form)
+      TestUtils.renderIntoDocument(
+        <Provider store={store}>
+          <TestForm/>
+        </Provider>
+      )
+
+      // username input rendered
+      expect(usernameInput).toHaveBeenCalled()
+      expect(usernameInput.calls.length).toBe(1)
+
+      // username field has warning
+      expect(usernameInput.calls[ 0 ].arguments[ 0 ].meta.warning).toBe('Recommended')
+
+      // update username field so it passes
+      usernameInput.calls[ 0 ].arguments[ 0 ].input.onChange('erikras')
+
+      // username input rerendered twice, once for value, once for sync warning
+      expect(usernameInput.calls.length).toBe(3)
+
+      // should be valid now
```

Comment:
just realized some of these comments are nonsensical now. I *could* re-add the validity checks (warning should not invalidate form), or just take out the comments.